server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -100,6 +122,170 @@ export class MetadataExtractionProcessor {
+      deviceId: 'NONE',
+
+      type: AssetType.VIDEO,
+      isFavorite: false,
+      isArchived: sourceAsset.isArchived,
+      duration: null,
+      isVisible: false,
+      livePhotoVideo: null,
+      resizePath: null,
+      webpPath: null,
+      thumbhash: null,
+      encodedVideoPath: null,
+      tags: [],
+      sharedLinks: [],
+      originalFileName: path.parse(video).name,
+      faces: [],
+      sidecarPath: null,
+      isReadOnly: sourceAsset.isReadOnly,
+    });
+
+    sourceAsset.livePhotoVideoId = liveAsset.id;
+    await this.assetRepository.save(sourceAsset);
+    return liveAsset;
+  }
+
+  private async extractNewPixelLivePhoto(
+    asset: AssetEntity,
+    directory: DirectoryEntry[],
+    fileCreatedAt: Date | null,
+  ): Promise<AssetEntity | null> {
+    if (asset.livePhotoVideoId) {
+      // Already extracted, don't try again.
+      const [ret] = await this.assetRepository.getByIds([asset.livePhotoVideoId]);
+      this.logger.log(`Already extracted asset ${ret.originalPath}.`);
+      return ret;
+    }
+    let foundMotionPhoto = false;
+    let motionPhotoOffsetFromEnd = 0;
+    let motionPhotoLength = 0;
+    let motionPhotoMime = '';
+
+    // Look for the directory entry with semantic label "MotionPhoto", which is the embedded video.
+    // Then, determine the length from the end of the file to the start of the embedded video.
+    for (const entry of directory) {
+      if (entry.Item.Semantic == 'MotionPhoto') {
+        if (foundMotionPhoto) {
+          this.logger.error(`Asset ${asset.originalPath} has more than one motion photo.`);
+          continue;
+        }
+        foundMotionPhoto = true;
+        motionPhotoLength = entry.Item.Length ?? 0;
+        motionPhotoMime = entry.Item.Mime;
+      }
+      if (foundMotionPhoto) {
+        motionPhotoOffsetFromEnd += entry.Item.Length ?? 0;
+        motionPhotoOffsetFromEnd += entry.Item.Padding ?? 0;
+      }
+    }
+
+    if (!foundMotionPhoto || motionPhotoLength == 0) {
+      return null;
+    }
+    return this.extractEmbeddedVideo(
+      asset,
+      motionPhotoOffsetFromEnd,
+      motionPhotoLength,
+      motionPhotoMime,
+      fileCreatedAt,
+    );
+  }
+
+  private async extractEmbeddedVideo(
+    asset: AssetEntity,
+    offsetFromEnd: number,
+    length: number | null,
+    mime: string,
+    fileCreatedAt: Date | null,
+  ) {
+    const file = await fs.promises.open(asset.originalPath).catch((err) => {
+      this.logger.error(`Failed to open asset ${asset.originalPath}: ${err}`);
+      return null;
+    });
+    if (!file) {
+      return null;
+    }
+    let extracted = null;
+    try {
+      // Read in embedded video.
+      const stat = await file.stat().catch((err) => {
+        this.logger.error(`Failed to stat asset ${asset.originalPath}: ${err}`);
+        return null;
+      });
+      if (!stat) {
+        return null;
+      }
+      if (length == null) {
+        length = offsetFromEnd;
+      }
```

Comment:
You can probably just use one try/catch for this entire block. If any error is thrown, log it and return null. This removes a bunch of your if null return null statements.

Reply:
Done.